SupportClient/SupportClient/Controllers/FeedbackViewController.swift

```diff
@@ -0,0 +1,219 @@
+        let cell = collectionView.dequeueReusableCell(withReuseIdentifier: ErrorFileCollectionViewCell.reuseID, for: indexPath) as! ErrorFileCollectionViewCell
+        let errorFile = attachedErrorFiles[indexPath.item]
+        cell.configure(for: errorFile)
+        return cell
+    }
+
+}
+
+// MARK: - UICollectionView Delegate
+extension FeedbackViewController: UICollectionViewDelegate {
+    
+    // TODO: Respond to user taps to open image editor for markup
+    
+}
+
+// MARK: - UICollectionView DelegateFlowLayout
+extension FeedbackViewController: UICollectionViewDelegateFlowLayout {
+    
+    func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, sizeForItemAt indexPath: IndexPath) -> CGSize {
+        let scaledScreenWidth = view.layer.bounds.width / ErrorFileCollectionViewCell.imageResizingRatio
+        let scaledScreenHeight = view.layer.bounds.height / ErrorFileCollectionViewCell.imageResizingRatio
+        return CGSize(width: scaledScreenWidth, height: scaledScreenHeight)
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, insetForSectionAt section: Int) -> UIEdgeInsets {
+        let collectionViewImagePadding = (UIScreen.main.bounds.width - UIScreen.main.bounds.width / ErrorFileCollectionViewCell.imageResizingRatio) / 2.0
+        return UIEdgeInsets(top: 0, left: collectionViewImagePadding, bottom: 0, right: collectionViewImagePadding)
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, minimumLineSpacingForSectionAt section: Int) -> CGFloat {
+        return 40
+    }
+
+}
+
+// MARK: - DZNEmptyDataSet
+extension FeedbackViewController: DZNEmptyDataSetSource, DZNEmptyDataSetDelegate {
+    
+    // If no screenshots/videos are added by the user, they are prompted to add files
+    func customView(forEmptyDataSet scrollView: UIScrollView!) -> UIView! {
+        
+        let customView = CustomUIView(width: view.layer.bounds.width, height: view.layer.bounds.height / 3)
+        customView.backgroundColor = UIColor.backgroundColor
+        
+        let noFileLabel = UILabel()
+        // Create ParagraphStyle to format label text
+        let paragraphStyle = NSMutableParagraphStyle()
+        paragraphStyle.alignment = .center
+        paragraphStyle.lineSpacing = 10
+        // Tell the user there are no images/videos to display
+        noFileLabel.lineBreakMode = .byWordWrapping
+        noFileLabel.numberOfLines = 0
+        let title = "No Attached Files\n"
+        let subtitle = "Add screenshots or screen recordings from your photo library if they help to describe the bug"
+        let titleAttributes: [NSAttributedString.Key : Any] = [
+            NSAttributedString.Key.font: UIFont._17RobotoMedium!,
```

Comment:
do you have to have the `!`?

Reply:
Yeah, or else I get the warning: Expression implicitly coerced from '[NSAttributedString.Key : Optional<NSObject>]' to '[NSAttributedString.Key : Any]'